Doc/library/re.rst

```diff
@@ -809,6 +831,13 @@ form.
       >>> print('|'.join(map(re.escape, sorted(operators, reverse=True))))
       /|\-|\+|\*\*|\*
 
+   This functions must not be used for the replacement string in :func:`sub`
+   and :func:`subn`, only backslashes should be escaped.  For example::
+
+      >>> int_re = r'\d+'
+      >>> print(re.sub('INT', int_re.replace('\\', r'\\'), r'INT(\.INT)?(e[+-]?INT)?'))
+      \d+(\.\d+)?(e[+-]?\d+)?
```

Comment:
I don't find this example particularly clear.  Why would someone want to use re.escape() on the replacement string?  Wouldn't using ``int_re = r'\\d+'`` (and a normal str.replace on INT) be easier?

Reply:
I tried to provide simplified real-word example. In Mailman re.sub() is used for creating a regular expression. They passed the pattern containing a \d as a replacement string and got an error when this became invalid. Someone could use re.escape() on the replacement string, because the replacement string looks similar to simple pattern (it expands \n and \1). And this will work while the replacement string don't contain other metacharacters except a backslash.

I'll replace this example with the better one.